src/utils/formHelperFunctions.ts

```diff
@@ -437,3 +437,35 @@ export function getInitialStep({
     (getOrigin as any)(steps).key
   );
 }
+
+export function castVal(servarType: string | undefined, val: any) {
+  // If there is no type it is a hidden field and we will treat it as a string
+  if (servarType === undefined) return String(val);
+  let castVal = val;
+  switch (servarType) {
+    case 'currency':
+    case 'integer_field':
+    case 'rating':
+    case 'slider':
+      castVal = Number(val);
+      break;
+    case 'checkbox':
+      castVal = val === 'true';
```

Comment:
or 'True'?

Reply:
actually might be better to default to truthy and otherwise case "false" and "False" to false